### PR TITLE
azure: Chat implementation

### PR DESCRIPTION
**Pull request by spolu**
Parity functionality between OpenAI and AzureOpenAI
Was also able to create azure backed data sources locally

I see ~900ms on average for embedding to Azure vs ~400ms on average for OpenAI (with some requests taking much longer these days)

This PR just brings Azure to parity, which could enable us to re-route to Azure at decently minimal effort.

### CODE REVIEW

**Comment by spolu** [comment]
Screenshot from 2023-11-22 10-34-39

File: core/src/providers/azure_openai.rs [hunk 1]
```diff
@@ -462,22 +464,113 @@ impl LLM for AzureOpenAILLM {
...
+                            _ => None,
+                        },
+                        None => None,
+                    },
+                )
+                .await?
+            }
+        };
+
+        // println!("COMPLETION: {:?}", c);
```
Comment (lasryaric) [inline core/src/providers/azure_openai.rs:560]: Do you want to keep that?

**Review by lasryaric — APPROVED** [review]
Little comment but LGTM otherwise.

File: core/src/providers/azure_openai.rs (hunk 1 above)
Comment (spolu) [reply core/src/providers/azure_openai.rs:560]: Useful for debugging

**Pull request merged**